app/frontend/src/features/auth/signup/CompleteSignupForm.tsx

```diff
@@ -319,17 +319,16 @@ export default function CompleteSignupForm() {
             />
             <div>
               <TOS />
-              <Button
-                classes={{
-                  label: authClasses.buttonText,
-                  root: authClasses.button,
-                }}
-                loading={loading}
-                onClick={() => setAcceptedTOS(true)}
-                disabled={acceptedTOS}
-              >
-                {acceptedTOS ? THANKS : ACCEPT}
-              </Button>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    onChange={() =>
+                      setAcceptedTOS((prevAcceptedTOS) => !prevAcceptedTOS)
```

Comment:
Wow, I can't believe I didn't know you could use the previous state 😆

Reply:
Stackoverflow to the rescue :D

Reply:
I'd say this is the preferred way to do set state if you need to rely on previous state (even though probably doesn't matter in this case, more getting into good habits :p), since React guarantees you always get back the correct previous state, rather than a small chance that this might read the "wrong state" if other things also set/changes the state.